grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandler.java

```diff
@@ -68,7 +69,14 @@ static UnframedGrpcErrorHandler ofJson() {
             } else {
                 message = ImmutableMap.of("grpc-code", grpcCode.name());
             }
-            final HttpStatus httpStatus = GrpcStatus.grpcStatusToHttpStatus(status);
+            final RequestLogAccess log = ctx.log();
+            final Throwable cause;
+            if (log.isAvailable(RequestLogProperty.RESPONSE_CAUSE)) {
```

Comment:
I think we probably are not able to get the response cause because the log is not complete.
I guess we need to use `status.getCause()`.

Reply:
>  I guess we need to use status.getCause().

I think `status.getCause()` is null always because it is rebuilt from trailers.
https://github.com/line/armeria/blob/dd9d74e298a888b2256c6bd687d8359441a82249/grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractUnframedGrpcService.java#L154

Reply:
Then, we can handle the only `Status` in `UnframedGrpcStatusFunction`? 🤔

Reply:
I guess so. We don't need the `cause` parameter from the first.

Reply:
Had a chat with minwoox. A response cause could be set after the completion of a response stream.
We can guarantee happen-before relationship by setting a response content before a stream close rather than closing in `closeListener()`.
The following links will be helpful for fixing the behavior. 
1. Close a response stream and close a listener
https://github.com/line/armeria/blob/cb66bcc22de260b0a56144b1cec2219dc373227a/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L423-L437
2. Set a response content in `closeListener()`
https://github.com/line/armeria/blob/cb66bcc22de260b0a56144b1cec2219dc373227a/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L587-L591
3. A response cause is set if a status is no-ok
https://github.com/line/armeria/blob/5b384fbe27e7e6f9225d6db91cbb684d09dfbb5e/grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcLogUtil.java#L57-L62

For example:
```java
private void doClose(Status status, Metadata metadata) {
    ...
    final HttpHeaders trailers = statusToTrailers(ctx, defaultTrailers, status, metadata);
    ctx.logBuilder().responseContent(GrpcLogUtil.rpcResponse(newStatus, firstResponse);
    try {
        ...
        res.close();
        ...
    } finally {
        closeListener(status, completed, false);
    }
}

private void closeListener(Status newStatus, boolean completed, boolean setResponseContent) {
    ...
    if (setResponseContent) {
        ctx.logBuilder().responseContent(GrpcLogUtil.rpcResponse(newStatus, firstResponse);
    } 
}
```

Reply:
Updated!